core/api/src/adapter.rs

```diff
@@ -177,4 +177,33 @@ where
     async fn peer_count(&self, ctx: Context) -> ProtocolResult<U256> {
         self.net.peer_count(ctx).map(Into::into)
     }
+
+    async fn get_storage_at(
+        &self,
+        ctx: Context,
+        address: H160,
+        position: Hash,
+        height: Option<u64>,
```

Comment:
```suggestion
    async fn get_storage_at(
        &self,
        ctx: Context,
        address: H160,
        position: Hash,
        state_root: H256,
```

```suggestion
    async fn get_storage_at(
        &self,
        ctx: Context,
        address: H160,
        position: Hash,
        state_root: H256,
    ) -> ProtocolResult<Bytes> {
```

IMO, the interface should build the MPT from the given root, and the process of getting the state root by `BlockID` should not be included here.